src/python/pants/util/dirutil.py

```diff
@@ -442,6 +442,23 @@ def touch(path, times=None):
     os.utime(path, times)
 
 
+def recursive_dirname(f):
+  """Given a relative path like 'a/b/c/d', yield all ascending path components like:
+
+        'a/b/c/d'
+        'a/b/c'
+        'a/b'
+        'a'
+        ''
+  """
+  prev = None
+  while f != prev:
+    yield f
+    prev = f
+    f = os.path.dirname(f)
+  yield ''
```

Comment:
was there an actual bug with the old impl here? that seemed cleaner and more readable to me.

Reply:
Yea: infinite loop for `.` as an input.